contributors/guide/pull-requests.md

```diff
@@ -0,0 +1,283 @@
+- [The Pull Request Submit Process](#the-pr-submit-process)
+  * [The Testing and Merge Workflow](#the-testing-and-merge-workflow)
+  * [Marking Unfinished Pull Requests](#marking-unfinished-pull-requests)
+  * [Comment Commands Reference](#comment-commands-reference)
+  * [Automation](#automation)
+  * [How the e2e Tests Work](#how-the-e2e-tests-work)
+- [Why was my pull request closed?](#why-was-my-pr-closed)
+- [Why is my pull request not getting reviewed?](#why-is-my-pr-not-getting-reviewed)
+- [Best Practices for Faster Reviews](#best-practices-for-faster-reviews)
+  * [0. Familiarize yourself with project conventions](#0-familiarize-yourself-with-project-conventions)
+  * [1. Is the feature wanted? Make a Design Doc or Sketch pull request](#1-is-the-feature-wanted-make-a-design-doc-or-sketch-pr)
+  * [2. Smaller Is Better: Small Commits, Small pull requests](#2-smaller-is-better-small-commits-small-prs)
+  * [3. Open a Different Pull Request for Fixes and Generic Features](#3-open-a-different-pr-for-fixes-and-generic-features)
+  * [4. Comments Matter](#4-comments-matter)
+  * [5. Test](#5-test)
+  * [6. Squashing and Commit Titles](#6-squashing-and-commit-titles)
+  * [7. KISS, YAGNI, MVP, etc.](#7-kiss-yagni-mvp-etc)
+  * [8. It's OK to Push Back](#8-its-ok-to-push-back)
+  * [9. Common Sense and Courtesy](#9-common-sense-and-courtesy)
+
+# Before You Submit a Pull Request
+
+This guide is for contributors who already have a pull request to submit. If you're looking for information on setting up your developer environment and creating code to contribute to Kubernetes, see the [development guide](development.md).
+
+First time contributors should head to the [Contributor Guide](/contributors/guide/README.md) to get started.
+
+**Make sure your pull request adheres to our best practices. These include following project conventions, making small pull requests, and commenting thoroughly. Please read the more detailed section on [Best Practices for Faster Reviews](#best-practices-for-faster-reviews) at the end of this doc.**
+
+## Run Local Verifications
+
+You can run these local verifications before you submit your pull request to predict the
+pass or fail of continuous integration.
+
+* Run and pass `make verify` (can take 30-40 minutes)
+* Run and pass `make test`
+* Run and pass `make test-integration`
+
+# The Pull Request Submit Process
+
+Merging a pull request requires the following steps to be completed before the pull request will be merged automatically.
+
+- [Sign the CLA](https://git.k8s.io/community/CLA.md) (prerequisite)
+- [Open a pull request](https://help.github.com/articles/about-pull-requests/)
+  - *For kubernetes/kubernetes repository only:* Add [release notes](/guide/release-notes.md) if needed.
+- Pass all e2e tests
+- Get all necessary approvals from reviewers and code owners 
+
+## The Testing and Merge Workflow
+
+The Kubernetes merge workflow uses comments to run tests and labels for merging pull requests.
```

Comment:
What do you think about something like:

"...uses special labels (link?) and commands(link https://prow.k8s.io/command-help) to trigger actions leading to merging pull requests."

Authoritative for labels might be https://github.com/kubernetes/test-infra/blob/master/label_sync/labels.yaml but that's not exactly human readable content.

Reply:
This labels and commands topic probably merits its own section.  That would help the TODO a few lines below.